packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts

```diff
@@ -445,6 +483,46 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
     });
   }
 
+  /**
+  * Creates peering connection role for acceptor VPC
+  */
+  public createAcceptorVpcRole(requestorAccountId: string): Role {
+    const peeringRole = new Role(this, 'VpcPeeringRole', {
+      assumedBy: new AccountPrincipal(requestorAccountId),
+      roleName: 'VpcPeeringRole',
+      description: 'Restrictive role for VPC peering',
+    });
+
+    peeringRole.addToPolicy(new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ['ec2:AcceptVpcPeeringConnection'],
+      resources: [`arn:${Aws.PARTITION}:ec2:${this.region}:${this.ownerAccountId}:vpc/${this.vpcId}`],
+    }));
+
+    peeringRole.addToPolicy(new PolicyStatement({
+      actions: ['ec2:AcceptVpcPeeringConnection'],
+      effect: Effect.ALLOW,
+      resources: [`arn:${Aws.PARTITION}:ec2:${this.region}:${this.ownerAccountId}:vpc-peering-connection/*`],
+      conditions: {
+        StringEquals: {
+          'ec2:AccepterVpc': `arn:${Aws.PARTITION}:ec2:${this.region}:${this.ownerAccountId}:vpc/${this.vpcId}`,
+        },
+      },
+    }));
```

Comment:
I did not see grant methods exist in the VPC interface, I may be wrong. I believe we need to have some grant methods, and refactor this function later to use the grant methods. I will leave this to @shikha372 to decide what is correct here.

Reply:
there is no method currently in VPC, but I agree we can model these permissions under grant method with something like grantPeeringConnection(grantee: iam.IGrantable) and account id as principal, i'll add it in.